pxtcompiler/emitter/service.ts

```diff
@@ -1254,22 +1248,31 @@ namespace ts.pxtc.service {
 . . . . .
 \``;
 
-    export function getSnippet(apis: ApisInfo, runtimeOps: pxt.RuntimeOptions, fn: SymbolInfo, n: ts.FunctionLikeDeclaration, python?: boolean): string {
+    export function getSnippet(apis: ApisInfo, takenNames: pxt.Map<SymbolInfo>, runtimeOps: pxt.RuntimeOptions, fn: SymbolInfo, decl: ts.FunctionLikeDeclaration, python?: boolean): string {
```

Comment:
We should consider refactoring this at some point; too many arguments

Reply:
I definitely agree. I spent a good hour this morning trying to unify lastApisInfo and this syntax info stuff and I had to bail, it was too tangle.

Reply:
I left a number of new TODO comments detailing some of the things I think are broken / inflexible with our current approach